nix/module.nix

```diff
@@ -0,0 +1,114 @@
+{ options, config, lib, pkgs, ... }:
+
+with lib;
+with lib.types;
+let
+  cfg = config.programs.fw-fanctrl;
+  fw-ectool = pkgs.callPackage ./packages/fw-ectool.nix {};
+  fw-fanctrl = pkgs.callPackage ./packages/fw-fanctrl.nix {};
+  defaultConfig = builtins.fromJSON (builtins.readFile ../config.json);
+in
+{
+  options.programs.fw-fanctrl = {
+    enable = mkOption {
+      type = bool;
+      default = false;
+      description = ''
+        Enable fw-fanctrl systemd service and install the needed packages.
+      '';
+    };
+    config = {
+      defaultStrategy = mkOption {
+        type = str;
+        default = defaultConfig.defaultStrategy;
+        description = "Default strategy to use";
+      };
+      strategyOnDischarging = mkOption {
+        type = str;
+        default = defaultConfig.strategyOnDischarging; 
+        description = "Default strategy on discharging";
+      };
+      batteryChargingStatusPath = mkOption {
+        type = str;
+        default = defaultConfig.batteryChargingStatusPath;
+        description = "";
```

Comment:
(Optional) The path to the file containing the actual battery charging status (defaults to "/sys/class/power_supply/BAT1/status")